### PR TITLE
[TASK] Improve ItemsProcFunc documentation

### PR DESCRIPTION
**Pull request by nhovratov**
- List all available parameters
- Replace flex with select in types
- Use the term method instead of function
- Use a styleguide example instead of a (more complex) core example
- Small text fixes

### CODE REVIEW

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 1]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
    :Scope: Display / Proc.
-   :Types: :ref:`check <columns-check>`, :ref:`flex <columns-flex>`,
-      :ref:`radio <columns-radio>`
+   :Types: :ref:`check <columns-check>`, :ref:`select <columns-select>`, :ref:`radio <columns-radio>`
 
-   PHP function which is called to fill / manipulate the array with elements.
+   PHP method which is called to fill or manipulate the items array.
 
-   [classname]->[methodname]
+   VENDOR\\Extension\\UserFunction\\YourClass->yourMethod
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:16]: In general it is recommended to use the actual FQCN with `::class` instead and then concatenate the method:

```php
\VENDOR\Extension\UserFunction\YourClass::class . '->yourMethod'
```

Big advantage: with an IDE or similar places like these are also updated on refactoring.

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 2]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
-                }
-             }
-          }
-       }
-
-   Further examples on different config type's can be found in the extension `styleguide`.
+   Passed parameters
+     * :php:`items` (passed by reference)
+     * :php:`config` (TCA config of the field)
+     * :php:`TSconfig`
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:25]:
```suggestion
     * :php:`TSconfig` (currently active page TSconfig)
```

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 3]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
+     * :php:`table` (current table)
+     * :php:`row` (current database record)
+     * :php:`field` (current field name)
 
+     .. versionadded:: 7.6
+     This property only exists if the field has a :ref:`flex parent <columns-flex>`.
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:31]:
```suggestion
        The following parameter only exists if the field has a :ref:`flex parent <columns-flex>`.
```

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 4]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
+
+     * :php:`flexParentDatabaseRow`
+
+     .. versionadded:: 11.2
+     These properties are filled if the current record has an :ref:`inline parent <columns-inline>`.
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:36]:
```suggestion
        The following parameters only exist if the current record has an :ref:`inline parent <columns-inline>`.
```

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 5]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
+     * :php:`inlineParentUid`
+     * :php:`inlineParentTableName`
+     * :php:`inlineParentFieldName`
+     * :php:`inlineParentConfig`
+     * :php:`inlineTopMostParentUid`
+     * :php:`inlineTopMostParentTableName`
+     * :php:`inlineTopMostParentFieldName`
+
+   Example
+     The configuration for the field `category_field` of the table `tt_content` looks like this:
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:47]:
```suggestion
     The configuration for a custom field :sql:`category_field` within the table :sql:`tt_content` could look like this:
```

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 6]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
+
+     .. code-block:: php
+
+        'type' => 'select',
+        'renderType' => 'selectSingle',
+        'itemsProcFunc' => 'TYPO3\CMS\Core\Category\CategoryRegistry->getCategoryFieldsForTable',
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:53]:
```suggestion
        'itemsProcFunc' => \TYPO3\CMS\Core\Category\CategoryRegistry::class . '->getCategoryFieldsForTable',
```

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 7]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
+
+     .. code-block:: php
+
+        /**
+         * Returns a list of category fields for a given table for populating selector "category_field"
+         * in tt_content table (called as itemsProcFunc).
+         *
+         * @param array $configuration Current field configuration
+         * @throws \UnexpectedValueException
+         * @internal
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:71]:
```suggestion
```

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst [hunk 8]
```diff
@@ -9,62 +9,86 @@ itemsProcFunc
...
+                 foreach ($fields as $fieldName => $options) {
+                    $fieldLabel = $this->getLanguageService()->sL($GLOBALS['TCA'][$tableName]['columns'][$fieldName]['label']);
+                    $configuration['items'][] = [$fieldLabel, $fieldName];
+                 }
+              }
+           }
+        }
+
+     .. note::
+        This example uses a non-existing TCA property :php:`itemsProcConfig` in order to pass an own argument
```
Comment (mbrodala) [inline Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:92]:
```suggestion
        This example uses a custom TCA option :php:`itemsProcConfig` in order to pass an own argument
```

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst (hunk 2 above)
Comment (nhovratov) [reply Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:25]: Actually it's not the page tsconfig but this: https://docs.typo3.org/m/typo3/reference-tsconfig/master/en-us/PageTsconfig/TceForm.html?highlight=itemsprocfunc#itemsprocfunc

Do you know how to interlink to this?
It starts with `<t3tsconfig:` but I don't know what comes after. Any idea?

Comment (mbrodala) [reply Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst:25]: You can actually see it in your link: `itemsprocfunc`. :-)

All IDs/anchors are actually used for linking.

See https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/Hyperlinks.html#from-one-manual-to-another

**Comment by linawolf** [comment]
We are trying to move all code examles in the TCA Reference to the styleguide extension. This way we have code in the examples that is ensured to be working and kept up to date. 

Did you have a look if there is already an example in the styleguide extension that we could use?

**Comment by nhovratov** [comment]
@linawolf Yes it has some examples! So should I pick one and copy it to here?

File: Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst (hunk 2 above)
Comment (nhovratov) [reply Documentation/ColumnsConfig/CommonProperties/ItemsProcFunc.rst]: Thank you. It seems most headers have a reference to them like `.. _itemsProcFunc:`. This is missing for this exact option.

**Comment by linawolf** [comment]
@nhovratov yes, that would be wonderful

**Comment by nhovratov** [comment]
@linawolf Did it! I have a question: There is now this new <dl> tag (for the blue background). And they can themselves have other <dt> tags (gray color). Are they supposed to be sub-properties of an TCA option? I put the headers like "Example" in those places, but I don't think it's semantically correct. How to go about this? If I use normal headers, the identation is gone of course. But as the options are now separated into own pages this should be no problem.

**Review by linawolf — APPROVED** [review]

**Comment by linawolf** [comment]
@mbrodala can this be merged?

**Comment by linawolf** [comment]
Merging this now, if you have more changes you can make another pr, @mbrodala

**Pull request merged**